src/redux/matrix/reducer.ts

```diff
@@ -0,0 +1,83 @@
+import types from './types';
+import {randomGenerator, matrixSort} from '../../function/index';
+import {MatrixActions, MatrixStore} from './interface';
+
+const initialState: MatrixStore = {
+  settings: {
+    columns: 0,
+    rows: 0,
+    cells: 0,
+  },
+  matrixRows: [],
+  sortedMatrix: [],
+  nearest: [],
+};
+
+export const matrix = (state = initialState, action: MatrixActions) => {
+  switch (action.type) {
+    case types.SET_SETTINGS:
+      return {
+        ...state.settings,
+        settings: action.payload.settings,
+        matrixRows: action.payload.matrix,
+        sortedMatrix: action.payload.sortedMatrix,
+      };
+
+    case types.CREATE_MATRIX:
+      return {...state, matrixRows: [action.payload]};
+
+    case types.INCREMENT_CELL:
+      const newMatrixRows = state.matrixRows.map((row) => {
+        return row.map((item) => {
+          if (item.ID === action.payload.ID) {
+            // item.Amount += 1;
+            return {...item, Amount: item.Amount + 1};
+          }
+          return item;
+        });
+      });
+      return {
+        ...state,
+        matrixRows: newMatrixRows,
+        sortedMatrix: [...matrixSort(newMatrixRows)],
+      };
+
+    case types.DELETE_ROW:
+      const newMatrix = [...state.matrixRows].splice(action.payload, 1);
+      return {
+        ...state,
+        sortedMatrix: [...matrixSort(newMatrix)],
+        //TODO
+        // ...state.sortedMatrix.filter((el) =>
+        //   state.matrixRows[action.payload].every((element) => element.ID !== el.ID)
+        // ),
```

Comment:
TODO what? Need to describe what you need to do in comment

Reply:
already nothing))))